compiler/damlc/tests/daml-test-files/DiscloseViaChoiceObserver.daml

```diff
@@ -0,0 +1,38 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+module DiscloseViaChoiceObserver where
+
+-- @SINCE-LF 1.dev
+
+-- This example demonstrates the canonical use of choice-observers to achieve disclosure.
+
+template Secret with
+    owner: Party
+    text: Text
+  where
+    signatory owner
+    nonconsuming choice DiscloseTo: () with
+        receiver: Party
+      observer receiver
+      controller owner
+      do pure ()
+    nonconsuming choice Reveal: Text with
+        receiver: Party
+      controller receiver
+      do pure text
+
+test : Scenario ()
+test = scenario do
```

Comment:
This checks only that we divulge the contract which contains the choice with an observer. We should also add a test for checking that we also divulge everything happening in the choice body. Does that make sense?

Reply:
You mean, for example, if the choice body fetches some other contracts, these also need to be divulged?

Reply:
I added another test.